Packs/WootCloud/Integrations/WootCloud/WootCloud.py

```diff
@@ -0,0 +1,255 @@
+        LOG(e)
+        return 'not ok'
+
+
+def fetch_single_alert(client, alert_id, type):
+    """ Fetches single packet by ID. """
+    if type == "packet":
+        url = 'packetalerts'
+    elif type == "bluetooth":
+        url = 'btalerts'
+    elif type == "anomaly":
+        url = 'anomalies'
+    else:
+        return_error('Type error: %s is not one of the types' % type)
+    return client.http_request('GET', 'events/%s/%s' % (url, alert_id))
+
+
+def fetch_incidents(client, alert_type):
+    """
+    fetches alerts and turns them into incidents.
+    """
+    last_run = demisto.getLastRun()
+    # Get the last fetch time, if exists
+    last_fetch = last_run.get('time')
+
+    # Handle first time fetch, fetch incidents retroactively
+    if last_fetch is None:
+        fetch_time, _ = parse_date_range(FETCH_TIME)
+        last_fetch = fetch_time.strftime(DATE_FORMAT)
+
+    incidents = []
+    ###
+    items = iter_all_alerts(client, alert_type, last_fetch, datetime.now().strftime(DATE_FORMAT),
+                            severity=SEVERITY_TYPE)
+    ###
+    for item in items:
+        incident = item_to_incident(item)
+        incident_date = datetime.strptime(incident['timestamp'], DATE_FORMAT)
+        # Update last run and add incident if the incident is newer than last fetch
+        if incident_date.timestamp() > datetime.strptime(last_fetch, DATE_FORMAT).timestamp():
+            last_fetch = incident_date.strftime(DATE_FORMAT)
+            incidents.append(incident)
+    demisto.setLastRun({'time': last_fetch})
+    demisto.incidents(incidents)
+
+
+''' COMMANDS MANAGER / SWITCH PANEL '''
+
+
+def main():
+    LOG('Command being called is %s' % (demisto.command()))
+
+    try:
+        client = Client(SERVER, SSL_VERIFY, API_VERSION, HEADERS, USERNAME, PASSWORD)
+
+        if demisto.command() == 'test-module':
+            # This is the call made when pressing the integration test button.
+            result = test_module(client)
+            return_results(result)
+        elif demisto.command() == 'fetch-incidents':
+            # Set and define the fetch incidents command to run after activated via integration settings.
+            fetch_incidents(client, ALERT_TYPE)
+        elif demisto.command() == 'wootcloud-get-pkt-alerts':
+            starttime, endtime = parse_date_range(demisto.args().get('date_range'), DATE_FORMAT)
+            alerts = client.get_woot_alerts('packet', starttime, endtime, severity=demisto.args().get('severity'),
+                                            skip=demisto.args().get('skip'), limit=demisto.args().get('limit'),
+                                            site_id=demisto.args().get('site_id'))
+            return_results(CommandResults(outputs=alerts['packet_alerts'],
+                                          outputs_prefix='WootCloud.PacketAlert',
+                                          outputs_key_field='id'))
+        elif demisto.command() == 'wootcloud-get-bt-alerts':
```

Comment:
The implementation of the command should be in a separate command that will return the CommandResult object.
In the main should be only the return_results.
https://xsoar.pan.dev/docs/integrations/code-conventions#main-function

Reply:
commited

Reply:
committed